### PR TITLE
Week 3 Step Four - Uday

### PR DESCRIPTION
**Pull request by UdayKalra**
Implement a comment submission feature. One can test by typing a comment in the field in index.html, submit comment, then fetch comments. Deploy at each step's completion at go/UdayKalra-app .

### CODE REVIEW

File: udaykalra/portfolio/src/main/webapp/style.css [hunk 1]
```diff
@@ -67,6 +62,11 @@
   --text-color: #e7e7e7;
 }
 
+#quote-container {
+  margin-top: 20px;
+  color: #fff;
```
Comment (aaronegolden) [inline udaykalra/portfolio/src/main/webapp/style.css:67]: Missing a few f's? #ffffff

Comment (UdayKalra) [reply udaykalra/portfolio/src/main/webapp/style.css:67]: GCloud condensed it to 3 f's as they represent the same color.

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -36,14 +36,43 @@ private String arrayListToJson(ArrayList<String> inputList) {
...
+    }
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+  * Obtains parameter from Comments typing field.
+  */
+  private String getParameter(HttpServletRequest request, String comment, String defaultValue) {
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:71]: Seems like `comment` would be more accurately named `parameter` or `parameterName`, since you're looking for a specific parameter in request (this could be a comment, but could also be the "erase" checkbox).

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -36,14 +36,43 @@ private String arrayListToJson(ArrayList<String> inputList) {
...
+
+  /**
+  * Writes comment data to ArrayList data
+  */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+    boolean eraser = Boolean.parseBoolean(getParameter(request, "erase", "false"));
+    comments.add(text);
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: Do you still want to add the comment if it's just an empty string?

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (ykjchen) [inline udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:71]: Nit: how about naming this `getRequestParameter`, since `getParameter` is fairly generic and you may also want to get a parameter from something else in the future, which would require disambiguating this method name.

File: udaykalra/portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -28,6 +28,7 @@ function getCommentData() {
         const commentsElement = document.getElementById('quote-container');
         commentsElement.innerHTML = '';
         for (var inc = 0; inc < myObject.length; inc += 1) {
+          console.log(myObject);
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/webapp/script.js:31]: What does `myObject` contain? I'm wondering if it's name can be made more specific, so the reader would understand without reading through the rest of this method what will be logged.

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -22,10 +22,10 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some comments.*/
```
[the remark this answers is not in the document]
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -36,14 +36,43 @@ private String arrayListToJson(ArrayList<String> inputList) {
...
   }
+
+  /**
+  * Writes comment data to ArrayList data
+  */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+    boolean eraser = Boolean.parseBoolean(getParameter(request, "erase", "false"));
```
[the remark this answers is not in the document]
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: udaykalra/portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/webapp/script.js]: Done

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: udaykalra/portfolio/pom.xml [hunk 7]
```diff
@@ -29,8 +29,8 @@
       <artifactId>gson</artifactId>
       <version>2.8.6</version>
     </dependency>
-  </dependencies>
-
+  </dependencies>  
+
```
Comment (ykjchen) [inline udaykalra/portfolio/pom.xml:33]: Revert these changes? Looks like some trailing whitespace was added.

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -37,14 +37,44 @@ private String arrayListToJson(ArrayList<String> inputList) {
...
+  /**
+   * Writes comment data to ArrayList data
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getRequestParameter(request, "text-input", "");
+    boolean toErase = Boolean.parseBoolean(getRequestParameter(request, "erase", "false"));
+
+    if (toErase) {
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:59]: `shouldErase` may sound more fluent. Compare: `if to erase` and `if should erase` when reading this condition.

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 9]
```diff
@@ -37,14 +37,44 @@ private String arrayListToJson(ArrayList<String> inputList) {
...
+      comments = new ArrayList<String>();
+    } else if (!(text.equals(""))) {
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:61]: How would you like to handle the text being " " (or any number of whitespaces)?

File: udaykalra/portfolio/src/main/webapp/script.js [hunk 10]
```diff
@@ -25,10 +25,11 @@ function createCommentList(inputText) {
 function getCommentData() {
   fetch('/data')                          // sends a request to /data
       .then(response => response.json())  // parses the response as JSON
-      .then((myComments) => {  // now we can reference the fields in myComments!
+      .then((myComments) => {  // now we can reference the fields in myObject!
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/webapp/script.js:28]: As suggested in the previous PR, consider renaming myComments -> comments (to increase conciseness at no cost to clarity)
